modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx

```diff
@@ -76,6 +76,21 @@ export function Attachments({setValues, values}: IProps) {
 		setAttachmentsFields(parseFields(fields));
 	};
 
+	useEffect(() => {
+		API.getAllObjectDefinitions().then((items) => {
+			const objectDefinitions = items.filter(({system}) => !system);
+
+			setObjectDefinitions(objectDefinitions);
+
+			const currentObjectDefinition = objectDefinitions?.find(
+				(item) => item.id === values.objectDefinitionId
+			);
+
+			setSelectedEntity(currentObjectDefinition);
+		});
```

Comment:
Hey @Artartur, to follow a pattern we use here we can use async/await notation, as useEffect by default cannot use async notation we can [write the async function inside useEffect](https://devtrium.com/posts/async-functions-useeffect#write-the-asynchronous-function-inside-the-useeffect) and call it at the end, also you can set the states at the end, it makes it easier to read the function, like:


```tsx
useEffect(() => {
  const makeFetch = async () => {
	  const objectDefinitions = await API.getAllObjectDefinitions();
  
	  const currentObjectDefinition = objectDefinitions?.find(
		  (item) => item.id === values.objectDefinitionId
	  );
  
	  setObjectDefinitions(objectDefinitions.filter(({system}) => !system));
	  setSelectedEntity(currentObjectDefinition);
  }
  
  makeFetch();
}, [values.objectDefinitionId]);
```